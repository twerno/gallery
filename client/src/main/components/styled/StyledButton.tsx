import styled, { css } from 'styled-components';

export interface IStyledButtonProps {
    mode: 'standalone' | 'inputButton';
}

export default styled.button<IStyledButtonProps>`
    padding: 9px 18px;
    background-color: transparent;
    color: black;
    border: 1px solid #cecece;
    border-radius: ${isStandalone('5px', '0')};
    border-left: ${isStandalone('', 'none')};

    font-size: 1rem;
    text-transform: uppercase;
    letter-spacing: 0.75px;

    &:focus {
        outline-color: rgb(103, 156, 226);
    }
`;

function isStandalone(trueStatement: string, falseStatement?: string) {
    return (props: IStyledButtonProps) =>
        props.mode === 'standalone'
            ? trueStatement
            : falseStatement || "unset";
}
