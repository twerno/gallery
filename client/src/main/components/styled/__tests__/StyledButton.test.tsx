import * as React from 'react';
import { componentBasicTests } from "test/componentBasicTests";
import StyledButton from '../StyledButton';

describe("StyledButton", () => {

    componentBasicTests("StyledButton distinct", <StyledButton mode='standalone' />);
    componentBasicTests("StyledButton inputButton", <StyledButton mode='inputButton' />);
});