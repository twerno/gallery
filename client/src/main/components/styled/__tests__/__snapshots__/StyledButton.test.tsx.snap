// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton [StyledButton distinct] Component basic tests matches the snapshot 1`] = `
.c0 {
  padding: 9px 18px;
  background-color: transparent;
  color: black;
  border: 1px solid #cecece;
  border-radius: 5px;
  font-size: 1rem;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.75px;
  -moz-letter-spacing: 0.75px;
  -ms-letter-spacing: 0.75px;
  letter-spacing: 0.75px;
}

.c0:focus {
  outline-color: rgb(103,156,226);
}

<button
  className="c0"
  mode="standalone"
/>
`;

exports[`StyledButton [StyledButton inputButton] Component basic tests matches the snapshot 1`] = `
.c0 {
  padding: 9px 18px;
  background-color: transparent;
  color: black;
  border: 1px solid #cecece;
  border-radius: 0;
  border-left: none;
  font-size: 1rem;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.75px;
  -moz-letter-spacing: 0.75px;
  -ms-letter-spacing: 0.75px;
  letter-spacing: 0.75px;
}

.c0:focus {
  outline-color: rgb(103,156,226);
}

<button
  className="c0"
  mode="inputButton"
/>
`;
